File: demo/models/FeeCharge.py
from django.db import models
from demo.models.FeeType import FeeType

#======================================================================
# 
# Encapsulates data for model FeeCharge
#
# @author your_name_here
#
#======================================================================

#======================================================================
# Class FeeCharge Declaration
#======================================================================
class FeeCharge (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	feeCode = models.CharField(max_length=200, null=True)
	amount = Money
	appliedOn = models.DateField(null=True)
	account = models.ForeignKey('Account', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	loanAccount = models.ForeignKey('LoanAccount', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	feeType = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in FeeType])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.feeCode
		str = str + self.amount
		str = str + self.appliedOn
		str = str + self.feeType
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "FeeCharge";
    
	def objectType(self):
		return "FeeCharge";

File: bankingonaspdotnet/Domain/FeeCharge.cs

using bankingonaspdotnet.Contracts;

namespace bankingonaspdotnet.Domain;

public class FeeCharge
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public virtual long? FeechargeId { get; set; }
    public virtual string? FeeCode { get; set; }
    public virtual Money? Amount { get; set; }
    public virtual DateOnly? AppliedOn { get; set; }
    public virtual Account? Account { get; set; }
    public virtual LoanAccount? LoanAccount { get; set; }
    public virtual FeeType? FeeType { get; set; }

    public static FeeCharge FromRequest(FeeChargeRequest request)
    {
        return new FeeCharge
        {
            Id = request.Id,
            FeeCode = request.FeeCode,
            Amount = request.Amount,
            AppliedOn = request.AppliedOn,
            FeeType = request.FeeType,
        };
    }
}
